packages/data-table-server/src/dataTableService/services/SurveyResponseDataTableService.ts

```diff
@@ -0,0 +1,126 @@
+  apiClient: TupaiaApiClient;
+};
+
+enum BaseFilters {
+  countryCodes = 'country.code',
+  surveyCodes = 'survey.code',
+  entityCodes = 'entity.code',
+  assessorNames = 'survey_response.assessor_name',
+  ids = 'survey_response.id',
+  outdated = 'survey_response.outdated',
+}
+
+type Params = {
+  ids: string[];
+  startDate?: Date;
+  endDate?: Date;
+  countryCodes: string[];
+  surveyCodes: string[];
+  entityCodes: string[];
+  assessorNames: string[];
+  outdated: boolean;
+};
+
+enum ResponseColumns {
+  surveyResponseId = 'survey_response.id',
+  surveyResponseAssessorName = 'survey_response.assessor_name',
+  surveyResponseDataTime = 'survey_response.data_time',
+  surveyResponseOutdated = 'survey_response.outdated',
+  surveyId = 'survey.id',
+  surveyName = 'survey.name',
+  surveyCode = 'survey.code',
+  entityName = 'entity.name',
+  countryName = 'country.name',
+}
+
+type SurveyResponseDataTableServiceResponse = {
+  [ResponseColumns.countryName]: Country['name'];
+  [ResponseColumns.entityName]: Entity['name'];
+  [ResponseColumns.surveyId]: Survey['id'];
+  [ResponseColumns.surveyName]: Survey['name'];
+  [ResponseColumns.surveyCode]: Survey['code'];
+  [ResponseColumns.surveyResponseId]: SurveyResponse['id'];
+  [ResponseColumns.surveyResponseAssessorName]: SurveyResponse['assessor_name'];
+  [ResponseColumns.surveyResponseDataTime]: SurveyResponse['data_time'];
+  [ResponseColumns.surveyResponseOutdated]: SurveyResponse['outdated'];
+};
+
+/**
+ * DataTableService for pulling data from aggregator's fetchDataGroup() endpoint
+ */
+export class SurveyResponseDataTableService extends DataTableService<
+  SurveyResponseDataTableServiceContext,
+  typeof requiredParamsSchema,
+  typeof configSchema,
+  SurveyResponseDataTableServiceResponse
+> {
+  protected supportsAdditionalParams = false;
+
+  public constructor(context: SurveyResponseDataTableServiceContext, config: unknown) {
+    super(context, requiredParamsSchema, configSchema, config);
+  }
+
+  protected getFilter(params: Params) {
+    const filter = Object.entries(BaseFilters).reduce((acc, [filterKey, filterField]) => {
+      const filterValue = params[filterKey as keyof Omit<Params, 'startDate' | 'endDate'>];
+
+      if (filterValue === undefined || filterValue === null) return acc;
+
+      if (Array.isArray(filterValue) && filterValue.length === 0) return acc;
+
+      return {
+        ...acc,
+        [filterField]: filterValue,
+      };
+    }, {} as Record<string, any>);
+
+    if (params.startDate && params.endDate) {
+      // Set the start date to the beginning of the day and the end date to the end of the day
+      const startDate = new Date(params.startDate).setHours(0, 0, 0, 0);
+      const endDate = new Date(params.endDate).setHours(23, 59, 59, 999);
+      filter['survey_response.data_time'] = {
+        comparator: 'between',
+        comparisonValue: [new Date(startDate), new Date(endDate)],
+      };
+    }
```

Comment:
I think we could have a couple more conditions here for when either startDate or endDate are present. The 3 options should be:
```
startDate: data_time > startDate
endDate: data_time < endDate
startDate && endDate: startDate  < data_time < endDate
```